src/ripple/app/tests/AmendmentTable.test.cpp

```diff
@@ -32,349 +33,299 @@ namespace ripple
-                    if (tablePopulationAlgo == TablePopulationAlgo::addInitial)
-                        expect (table->isEnabled (i.id ()));
-                }
+        for (auto const& a : m_set3)
+            expect (!table->find (a));
+        for (auto const& a : m_set4)
+            expect (!table->find (a));
+    }
 
-                for (auto const& i : unknownAmendmentNames)
-                {
-                    expect (!table->isSupported (i.id ()));
-                    expect (!table->isEnabled (i.id ()));
-                }
-            }
+    void testBadConfig ()
+    {
+        testcase ("Bad Config");
 
+        auto const section = makeSection (m_set1);
+        auto const id = to_string (amendmentId (m_set2[0]));
+
+        { // Two arguments are required - we pass one
+            Section test = section;
+            test.append (id);
+
+            try
             {
-                // check that we throw an exception on bad hex pairs
-                StringPairVec const badHexPairs (
-                    {{"a9f90e7cddbcadfedm8fc89ec4d02011c", "BadHex1"},
-                     {"c956ccabf25151a16d77T3171c485423b", "BadHex2"},
-                     {"6dcd528f057711c5d2Z6b57be28e23dfa", "BadHex3"}});
-
-                // make sure each element throws
-                for (auto const& i : badHexPairs)
-                {
-                    StringPairVec v ({i});
-                    auto table (makeTable (2));
-                    try
-                    {
-                        populateTable (*table, v, tablePopulationAlgo);
-                        // line above should throw
-                        fail ("didn't throw");
-                    }
-                    catch (...)
-                    {
-                        pass ();
-                    }
-                    try
-                    {
-                        populateTable (
-                            *table, badHexPairs, tablePopulationAlgo);
-                        // line above should throw
-                        fail ("didn't throw");
-                    }
-                    catch (...)
-                    {
-                        pass ();
-                    }
-                }
+                if (makeTable (2, test, emptySection, emptySection))
+                    fail ("Accepted only amendment ID");
+            }
+            catch (...)
+            {
+                pass();
             }
         }
 
-        {
-            // check that we thow on bad num tokens
-            std::vector<std::string> const badNumTokensConfigLines (
-                {"19f6d",
-                 "19fd6 bad friendly name"
-                 "9876 one two"});
-
-            // make sure each element throws
-            for (auto const& i : badNumTokensConfigLines)
+        { // Two arguments are required - we pass three
+            Section test = section;
+            test.append (id + "Test Name");
```

Comment:
I think this passes two arguments, not three (it probably passes the test because `Test` has non hex chars). We need to pass `" Test Name"` (notice the space before `Test`)